processing/src/main/java/org/apache/druid/segment/filter/cnf/HiveCnfHelper.java

```diff
@@ -178,6 +179,11 @@ private static void generateAllCombinations(
           a.add(child);
           // Result must receive an actual OrFilter.
           result.add(idempotentOr(Filters.or(a)));
+          if (result.size() >= 10_000) {
+            throw new CNFFilterExplosionException(
+                "Atleast 10,000 filters created by CNF (conjunctive normal form) conversion of the filter. "
```

Comment:
nit: Can `10_000` here and in the `if` be made a final variable somewhere? Would reduce the changes we need to track when updating the limit

Reply:
Yes, fixed by creating a constant